drivers/SmartThings/zigbee-window-treatment/src/micronic/init.lua

```diff
@@ -0,0 +1,117 @@
+      return true
+    end
+  end
+  return false
+end
+
+local function invert_preference_handler(device)
+  -- if invert is false then normal case handler or reverse case handler
+  local invert_value = device.preferences.reverse
+  local invert_cluster_cmd = cluster_base.write_manufacturer_specific_attribute(device, INVERT_CLUSTER,
+      INVERT_CLUSTER_ATTRIBUTE, 0x0000, data_types.Boolean, invert_value)
+  device:send(invert_cluster_cmd)
+end
+
+local function info_changed(driver, device, event, args)
+  if device.preferences ~= nil and device.preferences.reverse ~= args.old_st_store.preferences.reverse then
+    invert_preference_handler(device)
+  end
+end
+
+local set_window_shade_level = function(level)
+  return function(driver, device, cmd)
+    device:send_to_component(cmd.component, WindowCovering.server.commands.GoToLiftPercentage(device, level))
+  end
+end
+
+local function current_position_attr_handler(driver, device, value, zb_rx)
+
+  local level = value.value
+  local current_level = device:get_latest_state("main", capabilities.windowShadeLevel.ID,
+      capabilities.windowShadeLevel.shadeLevel.NAME)
+  --log.debug(string.format(" current_position_attr_handler   level=(%s) current level=(%s)", level, current_level))
+  local windowShade = capabilities.windowShade.windowShade
+  if level == 0 then
+    device:emit_event(windowShade.closed())
+    device:emit_event(capabilities.windowShadeLevel.shadeLevel(0))
+  elseif level == 100 then
+    device:emit_event(windowShade.open())
+    device:emit_event(capabilities.windowShadeLevel.shadeLevel(100))
+  else
+    if current_level ~= level or current_level == nil then
+      device:emit_event(windowShade.partially_open())
+    end
+  end
+end
+
+local function window_shade_level_cmd(driver, device, command)
+  local level = utils.clamp_value(command.args.shadeLevel, 0, 100)
+  device:send_to_component(command.component, WindowCovering.server.commands.GoToLiftPercentage(device, level))
+end
+
+local do_refresh = function(self, device)
+  device:send(WindowCovering.attributes.CurrentPositionLiftPercentage:read(device))
+  local invert_cluster_read = cluster_base.read_manufacturer_specific_attribute(device, INVERT_CLUSTER,
+      INVERT_CLUSTER_ATTRIBUTE, 0x0000)
+  device:send(invert_cluster_read)
+end
+
+local micronic_window_shade = {
+  NAME = "micronic window shade",
+  zigbee_handlers = {
+    attr = {
+      [WindowCovering.ID] = {
+        [WindowCovering.attributes.CurrentPositionLiftPercentage.ID] = current_position_attr_handler
+      }
+    }
+  },
+  capability_handlers = {
+    [capabilities.windowShade.ID] = {
+      [capabilities.windowShade.commands.open.NAME] = set_window_shade_level(100),
+      [capabilities.windowShade.commands.close.NAME] = set_window_shade_level(0)
+    },
+    [capabilities.windowShadeLevel.ID] = {
+      [capabilities.windowShadeLevel.commands.setShadeLevel.NAME] = window_shade_level_cmd
```

Comment:
If you want this to work, you'd need to include the `windowShadeLevel` capability in your profile.

If you just want to be able to open/close the shade, then you would delete this and also any `windowShadeLevel` events you are generating.